src/impls/$take/__tests__/$take.test.ts

```
@@ -10,4 +10,30 @@ describe($`take`, () => {
       expect($await($unwrap($take(2, $wrap([1, 2, 3]))))).toEqual([1, 2]);
     }),
   );
+  it(
+    'completes immediately if requesting 0 (or less) items',
+    $async(() => {
+      expect($await($unwrap($take(0, $wrap([1, 2, 3]))))).toEqual([]);
+    }),
+  );
+  it(
+    'completes immediately after taking the first n values',
+    $async(() => {
+      $async;
+      function* twoItemsThenNever() {
+        yield 1;
+        yield 2;
+        // Never yields 3.
```

One last change. Could we just throw an error here instead of causing the test runner to time out?

---

Yes, we could, but that case should never happens anyway as long as the implementation is correct. But you're the boss, on it